From b1a24bd22d63a2f7a4d120562e276b2165d2092d Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Wed, 2 Aug 2017 14:41:53 +0800
Subject: [PATCH 2273/5242] MLK-16121-2 arm64: dts: freescale: imx8mq: add
 over-drive mode support

commit  cde465abe38fb407212d8399c62c25b2867bd15c from
https://source.codeaurora.org/external/imx/linux-imx.git

i.MX8MQ can run at over-drive mode, it will need to increase
VDD_ARM voltage, add new dtb to support it, this feature
is only available on A1 board with GPIO1_IO13 to control
VDD_ARM voltage.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    3 +-
 .../arm64/boot/dts/freescale/fsl-imx8mq-evk-od.dts |   46 ++++++++++++++++++++
 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts   |    6 +++
 3 files changed, 54 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-od.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 40bbda3..1b1af67 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -28,4 +28,5 @@ dtb-$(CONFIG_ARCH_FSL_IMX8QXP) += fsl-imx8qxp-lpddr4-arm2.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-mlb.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-mqs.dtb \
 				  fsl-imx8qxp-lpddr4-arm2-wm8962.dtb
-dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += fsl-imx8mq-evk.dtb
+dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += fsl-imx8mq-evk.dtb \
+				 fsl-imx8mq-evk-od.dtb
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-od.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-od.dts
new file mode 100644
index 0000000..0f4574d
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-od.dts
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8mq-evk.dts"
+
+/ {
+	regulators {
+		reg_gpio_dvfs: regulator-gpio {
+			compatible = "regulator-gpio";
+			pinctrl-names = "default";
+			pinctrl-0 = <&pinctrl_dvfs>;
+			regulator-min-microvolt = <900000>;
+			regulator-max-microvolt = <1000000>;
+			regulator-name = "gpio_dvfs";
+			regulator-type = "voltage";
+			gpios = <&gpio1 13 GPIO_ACTIVE_HIGH>;
+			states = <900000 0x1 1000000 0x0>;
+		};
+	};
+};
+
+&pwm2 {
+	status = "disabled";
+};
+
+&A53_0 {
+	operating-points = <
+		/* kHz    uV */
+		1400000 1000000
+		1200000 900000
+		800000	900000
+	>;
+	dc-supply = <&reg_gpio_dvfs>;
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
index 47f71a3..7bd7788 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
@@ -141,6 +141,12 @@
 			>;
 		};
 
+		pinctrl_dvfs: dvfsgrp {
+			fsl,pins = <
+				MX8MQ_IOMUXC_GPIO1_IO13_GPIO1_IO13	0x16
+			>;
+		};
+
 		pinctrl_uart1: uart1grp {
 			fsl,pins = <
 				MX8MQ_IOMUXC_UART1_RXD_UART1_DCE_RX		0x79
-- 
1.7.9.5

